test/gtest/uct/test_pending.cc

```diff
@@ -328,6 +336,68 @@ UCS_TEST_P(test_uct_pending, pending_async)
     delete req;
 }
 
+/*
+ * test that arbiter does not block when ucs_ok is returned
+ * The issue is a dc transport specific but test may be also useful
+ * for other transports
+ */
+UCS_TEST_P(test_uct_pending, pending_ucs_ok_dc_arbiter_bug)
+{
+    ucs_status_t status;
+    ssize_t packed_len;
+    int N;
+    int i;
+    pending_send_request_t *req;
+
+    initialize();
+    check_caps(UCT_IFACE_FLAG_AM_BCOPY | UCT_IFACE_FLAG_PENDING);
+
+    mapped_buffer sbuf(ucs_min(64ul, m_e1->iface_attr().cap.am.max_bcopy), 0,
+                       *m_e1);
+
+    /* set a callback for the uct to invoke when receiving the data */
+    install_handler_sync_or_async(m_e2->iface(), 0, am_handler_simple, 0);
+
+    if (m_e1->iface_attr().cap.flags & UCT_IFACE_FLAG_CONNECT_TO_IFACE) {
+        N = 2048;
+    } else {
+        N = 128;
+    }
+
+    if (RUNNING_ON_VALGRIND) {
```

Comment:
what about merge this evaluation and previous one?
like
if (!VALGRIND) {
    if (CONNECTO_TO_IFACE) {
        N = 2048
    } else {
        N = 128
    }
} else {
    N = 64
}